test/images/agnhost/connect/connect.go

```diff
@@ -81,3 +95,26 @@ func main(cmd *cobra.Command, args []string) {
 	fmt.Fprintf(os.Stderr, "OTHER: %v\n", err)
 	os.Exit(1)
 }
+
+func connectSCTP(dest string, timeout time.Duration) {
+	addr, err := sctp.ResolveSCTPAddr("sctp", dest)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "DNS: %v\n", err)
+		os.Exit(1)
+	}
+
+	conn, err := sctp.DialSCTPExt("sctp", nil, addr, sctp.InitMsg{MaxInitTimeout: uint16(timeout.Milliseconds())})
```

Comment:
This doesn't seem to be correct; `MaxInitTimeout` → `sinit_max_init_timeo` is not a timeout on the connect operation as a whole, it's an upper bound on what the timeout can be for each attempt, of which there are 8 by default, starting at 3s each and then doubling until they hit the max. (eg, if you pass `--timeout 15` you'll get 3s + 6s + 12s + 15s + 15s + 15s + 15s + 15s = 1m36s).

There doesn't seem to be any good way to make the operation itself time out when you want it to without breaking the protocol so I think you're going to need to rewrite this to just do a timeout-less connect in a goroutine, with a separate timer.

Reply:
Thanks for pointing out, I completely mis-interpreted it. I'll go with the goroutine approach.